### DISCUSSION TITLE
POST request in `_middleware.ts`

### DISCUSSION THREAD

**Discussion by garrett-thompson** [vercel/next.js#34890] [Help] [answered]
In my middleware function, I need to make a POST request to a GraphQL server. Based on the response, the user may or may not have access to view the page they're requesting. 

I don't want to use `getServerSideProps` because this auth logic applies to entire route subtrees. For example, if we query the GraphQL API and the user has access to `/a` then they have access to every page under `/a` as well. That would get pretty unwieldy to have to implement the same check in `getServerSideProps` for all of those pages. Middleware is a much better option with it's behavior.

I'm aware of the latency repercussions of making this request, but unfortunately it's necessary for my usecase.

Specifically, I'm running into an error with the request body.

<img width="314" alt="Screen Shot 2022-02-28 at 1 26 11 PM" src="https://user-images.githubusercontent.com/9908831/156037709-b4f9462c-8c8c-4aa7-8042-616a2b492b59.png">
<img width="658" alt="Screen Shot 2022-02-28 at 1 26 43 PM" src="https://user-images.githubusercontent.com/9908831/156037719-16138020-30e4-4c1f-bbee-d86a80fab2ff.png">

**Comment by wwwdepot** [comment, marked as answer]
You can pass an Request object (you get this from another request) or URL string in the first arg. See https://developer.mozilla.org/en-US/docs/Web/API/Request/Request

We are not re-using previous request, so we pass a string in the first argument

```js
  const res = await fetch("http://api.example.com", {
    method: "POST",
    body: JSON.stringify(...),
  });
```
